kafka-connect/kafka-connect/src/test/java/org/apache/iceberg/connect/data/RecordConverterTest.java

```diff
@@ -921,11 +929,24 @@ private void assertRecordValues(Record record) {
     assertThat(rec.getField("dec")).isEqualTo(DEC_VAL);
     assertThat(rec.getField("s")).isEqualTo(STR_VAL);
     assertThat(rec.getField("b")).isEqualTo(true);
-    assertThat(rec.getField("u")).isEqualTo(UUID_VAL);
-    assertThat(rec.getField("f")).isEqualTo(BYTES_VAL);
+    assertThat(rec.getField("u")).isEqualTo(UUIDUtil.convert(UUID_VAL));
+    assertThat(rec.getField("f")).isEqualTo(BYTES_VAL.array());
     assertThat(rec.getField("bi")).isEqualTo(BYTES_VAL);
     assertThat(rec.getField("li")).isEqualTo(LIST_VAL);
     assertThat(rec.getField("ma")).isEqualTo(MAP_VAL);
+
+    // check by actually writing it
+    for (String format : new String[] {"parquet"}) {
+      IcebergSinkConfig config = mock(IcebergSinkConfig.class);
+      when(config.tableConfig(any())).thenReturn(mock(TableSinkConfig.class));
+      when(config.writeProps()).thenReturn(ImmutableMap.of("write.format.default", format));
+      WriteResult result = writeTest(ImmutableList.of(rec), config, UnpartitionedWriter.class);
+
+      assertThat(result.dataFiles()).hasSize(1);
+      assertThat(result.dataFiles())
+          .allMatch(file -> file.format() == FileFormat.fromString(format));
+      assertThat(result.deleteFiles()).hasSize(0);
```

Comment:
This tests fails before the fix for parquet and aparently fixing parquet it made ORC fail due to UUID handling differently